internal/pkg/service/common/etcdop/watchmirror.go

```diff
@@ -75,36 +80,43 @@ func SetupMirror[T any, V any](
 	}
 }
 
-func (s MirrorSetup[T, V]) StartMirroring(ctx context.Context, wg *sync.WaitGroup) (mirror *Mirror[T, V], initErr <-chan error) {
-	mirror = &Mirror[T, V]{
+func (s MirrorSetup[T, V]) Build() *Mirror[T, V] {
+	return &Mirror[T, V]{
```

Comment:
I have to create separate methods: `MirrorSetup.Build` (maybe it should be `MirrorBuilder`, rename it if you want), `Mirror.StartMirroring`, see next comments.